src/App.js

```diff
@@ -1,75 +1,202 @@
+        id,
+      })),
+    }));
+  }
+
+  changeActiveFilter = (filter) => {
+    this.setState({
+      activeFilter: filter,
+    });
+  }
 
-      <footer className="footer" style={{ display: 'block' }}>
-        <span className="todo-count">
-          3 items left
-        </span>
-
-        <ul className="filters">
-          <li>
-            <a href="#/" className="selected">All</a>
-          </li>
-
-          <li>
-            <a href="#/active">Active</a>
-          </li>
-
-          <li>
-            <a href="#/completed">Completed</a>
-          </li>
-        </ul>
-
-        <button
-          type="button"
-          className="clear-completed"
-          style={{ display: 'block' }}
+  removeCheckedTodos = () => {
+    this.setState(({ listOfTodos }) => ({
+      listOfTodos: listOfTodos.map(
+        todo => (!todo.isChecked ? todo : undefined)
+      ).filter(todo => todo !== undefined),
+    }));
+  }
+
+  toggleToEditingMode = (id, value) => {
+    this.setState(({ listOfTodos }) => ({
+      listOfTodos: listOfTodos.map(
+        todo => (todo.id === id
+          ? {
+            title: todo.title,
+            isChecked: todo.isChecked,
+            isEditing: !todo.isEditing,
+            id: todo.id,
+          }
+          : todo
+        )
+      ),
+      valueOfEditingInput: value,
+    }));
+  }
+
+  handleEditing = (value) => {
+    this.setState({
+      valueOfEditingInput: value.replace(/^\s+/, ''),
+    });
+  }
+
+  handleEditedSubmit = (id) => {
+    this.setState(({ listOfTodos, valueOfEditingInput }) => ({
+      listOfTodos: listOfTodos.map(
+        todo => (todo.id === id
+          ? {
+            title: valueOfEditingInput,
+            isChecked: todo.isChecked,
+            isEditing: !todo.isEditing,
+            id: todo.id,
+          }
+          : todo)
+      ),
+      valueOfEditingInput: '',
+    }));
+  };
+
+  render() {
+    const {
+      listOfTodos,
+      valueOfMainInput,
+      activeFilter,
+      valueOfEditingInput,
+    } = this.state;
+
+    const filteredTodos = activeFilter === 'active'
+      ? listOfTodos.filter(todo => !todo.isChecked)
+      : listOfTodos.filter(todo => todo.isChecked);
+
+    return (
+      <section className="todoapp">
+        <header className="header">
+          <h1>todos</h1>
+
+          <input
+            className="new-todo"
+            placeholder="What needs to be done?"
+            maxLength={100}
+            value={valueOfMainInput}
+            onChange={event => this.handleChangeMainInput(event.target.value)}
+            onKeyDown={e => this.handleSubmit(e.keyCode)}
+          />
+        </header>
+
+        <section className="main" style={{ display: 'block' }}>
+          {listOfTodos.length > 0 && (
+            <>
+              <input
+                type="checkbox"
+                id="toggle-all"
+                className="toggle-all"
+                checked={listOfTodos.every(todo => todo.isChecked)}
```

Comment:
better to put this value in state and change it with the func on the next line
because now you iterate the array after each rendering